uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1017.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1017 {
+
+  /**
+   * Migrating Phone Types and Deleting unwanted Phone Types
+   */
+  public function upgrade_1017() {
+    $this->migratePhoneTypes();
+    $this->relabelPhoneType("phone", "Landline");
+    $this->reservePhoneTypes(["phone", "mobile"]);
+    $this->deletePhoneTypes(["fax", "pager", "voicemail"]);
+    return TRUE;
+  }
+
+
+  /**
+   * Migrate the Phone Type Data as Follows:
+   * Fax -> LandLine
+   * Pager ->Mobile
+   * Voicemai -> Mobile
+   */
+  private function migratePhoneTypes() {
```

Comment:
Not sure of the performance implication of this like you mentioned in the PR description as per it being slow on client sites with thousands of phone registered. Wondering if we can use a different approach. We can use SQL to update the tables rather than using the API's We would get the id's if the original type and id of the new type and then in the loop we would run a query similar to
```php
UPDATE civicrm_phone SET phone_type_id = $newTypeID WHERE phone_type_id = $oldTypeID
```
This is guaranteed to run faster but we should discuss this.
@mickadoo , What do you think?

Reply:
@tunbola I had the same discussion about this with Davi [here](https://github.com/compucorp/civihr/pull/2637#discussion_r191777651). I agree though that a single query would be a lot better. At that time I said to Davi I could change it, but we decided to stick with API calls. I feel the same here. If we had some figures for the maxium number of phones on any client sites it would make it easier to decide.